src/lib/storage/mvcc_data.cpp

```diff
@@ -26,8 +26,12 @@ void MvccData::shrink() {
   tids.shrink_to_fit();
   begin_cids.shrink_to_fit();
   end_cids.shrink_to_fit();
+
+  _is_shrunk = true;
 }
 
+bool MvccData::is_shrunk() { return _is_shrunk; }
```

Comment:
Did you check if shrinking has any cost on a vector that is already shrunk?

Reply:
On a 100'000 size_t vector, shrink() takes ~2000ns and the following shrink()'s take <200ns.
So there is an overhead, but probably not enough. Will remove it.

Code:
```
BENCHMARK_F(MicroBenchmarkBasicFixture, TBBShrink)(benchmark::State& state) {
  tbb::concurrent_vector<size_t> values;

  for (auto i = size_t{0}; i < 100'000; ++i) {
    values.push_back(i);
  }

  auto start_time = std::chrono::high_resolution_clock::now();
  values.shrink_to_fit();
  auto end_time = std::chrono::high_resolution_clock::now();
  std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end_time - start_time).count() << std::endl;;

  for (auto i = size_t{0}; i < 5; ++i) {
    start_time = std::chrono::high_resolution_clock::now();
    values.shrink_to_fit();
    end_time = std::chrono::high_resolution_clock::now();
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end_time - start_time).count() << std::endl;;
  }
}
```